lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/io/ByteBuffers.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.io;
+
+import java.nio.ByteBuffer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public final class ByteBuffers
+{
+    private ByteBuffers() {}
+
+    /**
+     * Gets the bytes the provided {@link ByteBuffer} wraps, without advancing buffer position.
+     * Throws when provided buffer does not directly wrap bytes.
+     */
+    public static byte[] getWrappedBytes(ByteBuffer byteBuffer)
```

Comment:
It's my understanding that any modifications to the array returned by this method will also modify the backing array for the ByteBuffer. Are there any concerns with subsequent calls to this method failing due to the read position being modified?

Reply:
the read position isn't being modified. Added a test.